2_process/src/process.R

```diff
@@ -19,44 +14,42 @@ prep_data <- function(raw_data_file){
       model_type == 'dl' ~ 22,
       model_type == 'pgdl' ~ 23
     ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+  return(eval_data)
 }
 
 
 #' Save the processed data
 #' @param eval_data data table to save to csv
-#' @param out_filename string file name under which to save data locally (in out dir)
+#' @param file string file path under which to save data locally (in out dir)
 
-save_data <- function(eval_data, out_filename){
-  project_output_dir =  dirname(out_filename)
+write_csv <- function(eval_data, file){
```

Comment:
Just a note that I typically try to avoid naming functions I write the same name as existing package functions (e.g., `readr::write_csv`), as it can be confusing for anyone reading the code, particularly if you ended up adding additional steps into that function that deviate from what `readr::write_csv` would do.

Also - this approach (of calling a function that writes the csv and returns the filepath) works just fine, but for targets that only require a simple action like writing an existing dataframe to csv, you can actually include the steps to build the target in the target definition. It can feel a bit clunky for file targets, since you need to return the filepath, but it does work, and can be handy for small steps (e.g. filtering a dataframe):
```r
# Save the processed data
  tar_target(
    model_summary_results_csv,
    {
      out_file = "2_process/out/model_summary_results.csv"
      write_csv(eval_data, 
                file = out_file)
      return(out_file)
    }, 
    format = "file"
  ),
```

Reply:
If I include my return in my targets definition, do I no longer need to return anything in the function definition?

Also, is this relevant when the return is unrelated to the code carried out in the function? For example, I can use this method when I have a known filepath string as my return, but I cannot use this if my return in a variable/object/etc. that was created inside the function?

Reply:
Whoops I think my code was a bit unclear there b/c I was using `write_csv` which was the same name as your function. If you take the approach I showed above, you're not actually calling any function, you're just writing the code needed to build the target right there, with a call to `readr::write_csv`:
```r
# Save the processed data
  tar_target(
    model_summary_results_csv,
    {
      out_file = "2_process/out/model_summary_results.csv"
      readr::write_csv(eval_data, 
                file = out_file)
      return(out_file)
    }, 
    format = "file"
  ),
```
As for your second question, I think I'm a bit confused by what you mean. I'm having trouble thinking of a scenario when the return wouldn't be related to the code carried out in the function. In the case of a file target, we'll always be returning the filepath name, whether we pass that in as an argument, or define that within the function. We could do that in a function or right here in the code we're specifying to build the target:
```r
  tar_target(
    model_summary_results_csv,
    {
      out_file = sprintf("2_process/out/model_summary_results_%s.csv", eval_data$exper_id[1]) # pull random value into filename
      readr::write_csv(eval_data, 
                file = out_file)
      return(out_file)
    }, 
    format = "file"
  ),
```
Alternatively, if this was an object target, we could go ahead and modify and return an object without calling a function, for example if we wanted to filter `eval_data`:
``` r
  tar_target(
    eval_data_filtered,
    {
      eval_data %>% filter(exper_n = 1)
    }
  ),
```

Reply:
I think my second question no longer makes sense--I didn't realize you were defining the entire function in the targets, but I understand now--thanks for the clarification!